frontend/components/badge-examples.js

```diff
@@ -29,19 +29,22 @@ const Badge = ({
     baseUri,
     { longCache: false });
 
-  return (
-    <tr className={classNames({ excluded: !shouldDisplay() })}>
-      <th className={classNames({ clickable: onClick })} onClick={handleClick}>
-        { title }:
-      </th>
-      <td>{ previewImage }</td>
-      <td>
-        <code className={classNames({ clickable: onClick })} onClick={handleClick}>
-          { resolvedExampleUri }
-        </code>
-      </td>
-    </tr>
-  );
+  if (shouldDisplay()) {
+    return (
+      <tr>
+        <th className={classNames({ clickable: onClick })} onClick={handleClick}>
+          { title }:
+        </th>
+        <td>{ previewImage }</td>
+        <td>
+          <code className={classNames({ clickable: onClick })} onClick={handleClick}>
+            { resolvedExampleUri }
+          </code>
+        </td>
+      </tr>
+    );
+  }
+  return null;
```

Comment:
This makes the search more efficient because it only loads the badges matching our query (instead of hiding the ones that aren't relevant), but in isolation, this still background loads a load of badges when the user types one key (just because a lot of examples match the query), so elsewhere I've also restricted the search on the home page so that it only starts searching after the user enters 2 or more characters.

For the category-level search I haven't restricted it because we've already loaded all the badges in that category before we start searching.